backend/src/BinaryInstaller.ts

```diff
@@ -66,28 +67,34 @@ class BinaryInstaller {
 
   async uninstall(installers: Installer[]) {
     return new Promise(async (resolve, reject) => {
-      const commands = new Command({ admin: true, onError: reject })
-
-      if (Environment.isWindows) {
-        installers.map(installer => commands.push(`del /Q /F "${installer.binaryPath}"`))
-        if (existsSync(Environment.userPath)) commands.push(`rmdir /Q /S "${Environment.userPath}"`)
-        if (existsSync(Environment.adminPath)) commands.push(`rmdir /Q /S "${Environment.adminPath}"`)
-        if (existsSync(Environment.binPath)) commands.push(`rmdir /Q /S "${Environment.binPath}"`)
-      } else {
-        if (existsSync(Environment.userPath)) commands.push(`rm -rf ${Environment.userPath}`)
-        if (existsSync(Environment.adminPath)) commands.push(`rm -rf ${Environment.adminPath}`)
-        installers.map(installer => {
-          const files = installer.dependencyNames.concat(installer.binaryName)
-          files.map(
-            file => installer.fileExists(file) && commands.push(`rm -f ${path.join(Environment.binPath, file)}`)
-          )
-        })
-      }
+      const admin: boolean = true
 
+      // USER FILES
+      let commands = new Command({ onError: reject })
+      installers.map(i => this.removeFile(commands, i.binaryPath()))
+      this.removeDir(commands, environment.userPath)
       await commands.exec()
+
+      // ADMIN FILES - Should only be here if installed target by entering password
+      let adminCommands = new Command({ admin, onError: reject })
+      installers.map(i => this.removeFile(adminCommands, i.binaryPath(admin)))
+      this.removeDir(adminCommands, environment.adminPath)
+      if (environment.isWindows) this.removeDir(adminCommands, environment.binPath(admin))
+      await adminCommands.exec()
+
       resolve()
     })
   }
+
+  removeFile(commands: Command, path: string) {
+    if (!existsSync(path)) return
+    commands.push(environment.isWindows ? `del /Q /F "${path}"` : `rm -f ${path}`)
+  }
+
+  removeDir(commands: Command, path: string) {
+    if (!existsSync(path)) return
+    commands.push(environment.isWindows ? `rmdir /Q /S "${path}"` : `rm -rf ${path}`)
```

Comment:
`rm -rf` is **VERRRRRY** dangerous, I would be pretty cautious with such code. Consider using a library instead like rimraf or similar.

Reply:
Yea I actually did a full implementation with an npm module called trash, that would just move stuff to the trash but I couldn't get it to work. I would be happy to change this to a rimraf module, there might be an issue with running it with the sudo-prompt though. I'll poke around.